### PR TITLE
WebGLRenderer: Fix WebXR depth sensing.

### PR DESCRIPTION
**Pull request by cabanier**
Depth sensing is broken in the current version of three.js. 
This change will restore depth sorting functionality and also removes the extra renderpass of the previous solution.

The samples also didn't add the depthSensing dictionary during session creation. Quest browser wasn't spec compliant so didn't pass this in. That will be fixed in a future version of the spec.

cc @Mugen87

### CODE REVIEW

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1141,6 +1141,8 @@ class WebGLRenderer {
 
 			renderListStack.push( currentRenderList );
 
+			if ( _this.xr.getDepthSensingMesh() ) projectObject( _this.xr.getDepthSensingMesh(), camera, - Infinity, _this.sortObjects );
```
Comment (Mugen87) [inline src/renderers/WebGLRenderer.js:1144]: Can we write this like so:
```js
if ( xr.enabled === true && xr.isPresenting === true ) {

    const depthSensingMesh = _this.xr.getDepthSensingMesh();

    if ( depthSensingMesh !== null ) {

        projectObject( depthSensingMesh, camera, - Infinity, _this.sortObjects );

    }

}

Comment (cabanier) [reply src/renderers/WebGLRenderer.js]: done

**Review by Mugen87 — APPROVED** [review]

File: examples/webxr_xr_ballshooter.html [hunk 2]
```diff
@@ -81,7 +81,10 @@
 				controls.target.y = 1.6;
 				controls.update();
 
-				document.body.appendChild( XRButton.createButton( renderer, { 'optionalFeatures': [ 'depth-sensing'] } ) );
+				document.body.appendChild( XRButton.createButton( renderer, {
+					'optionalFeatures': [ 'depth-sensing' ],
+					'depthSensing': { 'usagePreference': [ 'gpu-optimized' ], 'dataFormatPreference': [] }
```
Comment (mrdoob) [inline examples/webxr_xr_ballshooter.html:86]: Is `'dataFormatPreference': []` needed?

Comment (cabanier) [reply examples/webxr_xr_ballshooter.html:86]: It is. We're updating the spec to specify that an empty array means that the value is optional.

**Pull request merged**